cli/azd/cmd/version.go

```diff
@@ -10,20 +10,42 @@ import (
 	"github.com/azure/azure-dev/cli/azd/internal"
 	"github.com/azure/azure-dev/cli/azd/pkg/commands"
 	"github.com/azure/azure-dev/cli/azd/pkg/environment"
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
 	"github.com/spf13/cobra"
 )
 
 func versionCmd(rootOptions *commands.GlobalCommandOptions) *cobra.Command {
-	return commands.Build(
-		commands.ActionFunc(
-			func(context.Context, *cobra.Command, []string, *environment.AzdContext) error {
-				fmt.Printf("azd version %s\n", internal.Version)
-				return nil
-			},
-		),
+	cmd := commands.Build(
+		commands.ActionFunc(versionAction),
 		rootOptions,
 		"version",
 		"Print the version number of azd",
 		"",
 	)
+
+	return output.AddOutputParam(
+		cmd,
+		[]output.Format{output.JsonFormat, output.NoneFormat},
+		output.NoneFormat,
+	)
+}
+
+func versionAction(_ context.Context, cmd *cobra.Command, _ []string, _ *environment.AzdContext) error {
+	formatter, err := output.GetFormatter(cmd)
+	if err != nil {
+		return err
+	}
+
+	switch formatter.Kind() {
+	case output.NoneFormat:
+		fmt.Printf("azd version %s\n", internal.Version)
+	case output.JsonFormat:
+		versionSpec := internal.GetVersionSpec()
```

Comment:
There was some discussion at one point about extending `version` to print version numbers of other tools we depend on.

Perhaps for safety right now we should instead format this as an object that looks like:

```
{
  "azd": "<version-number>"
}
```

So if we go add more versions later, we don't break consumers?

Reply:
@ellismg I like that. How about combining your and mine idea and having something like this

```json
{ 
  "azd": {
    "version": "1.2.3",
     "commit": "12345abc"
  }
}
```

Reply:
Pushed a related change--LMK what you think!

Reply:
Love it!